File: iOS/Sources/VertuEdge/HuggingFace/ModelInfo.swift
import Foundation

struct ModelInfo: Identifiable {
    let id: String
    let name: String
    let description: String
    let sizeStr: String?
    let downloadUrl: String
    let tags: [String]
}

enum DownloadState {
    case idle
    case downloading(modelName: String, progress: Float)
    case completed(modelName: String, localPath: String)
    case error(message: String)
}

File: Android/app/src/main/java/com/vertu/edge/huggingface/ModelInfo.kt
package com.vertu.edge.huggingface

data class ModelInfo(
    val id: String,
    val name: String,
    val description: String,
    val sizeStr: String? = null,
    val downloadUrl: String,
    val tags: List<String> = emptyList()
)

sealed class DownloadState {
    data object Idle : DownloadState()
    data class Downloading(val modelName: String, val progress: Float) : DownloadState()
    data class Completed(val modelName: String, val localPath: String) : DownloadState()
    data class Error(val message: String) : DownloadState()
}
